data-products/src/freestar_revenue_reporting/freestar_reporting_flow.py

```diff
@@ -0,0 +1,286 @@
+            "offset": 0,  # Start with an offset of 0
+        }
+    }
+
+    # Set up the headers with the API token
+    headers = {
+        "Authorization": f"Bearer {FREESTAR_API_KEY}",
+        "Content-Type": "application/json",
+    }
+
+    page_number = 1  # Initialize the page number tracker
+    total_count = 0  # Initialize the total count tracker
+
+    # create directory
+    Path(output_parquet_path).mkdir(parents=True, exist_ok=True)
+
+    while True:
+        # Send the POST request to the API with the current pagination parameters
+        response = requests.post(
+            API_BASE_URL + API_ENDPOINT, json=request_payload, headers=headers
+        )
+        response.raise_for_status()  # Raise an exception for HTTP errors
+
+        # Parse the JSON response
+        response_json = response.json()
+        logger.info(
+            f"Retrieved {len(response_json['data'])} records for page {page_number}."
+        )
+
+        # Remove the 'NdrPrebid' prefix from keys in the JSON response
+        for item in response_json["data"]:
+            for key in list(item.keys()):
+                if key.startswith("NdrPrebid."):
+                    new_key = key[len("NdrPrebid.") :]
+                    item[new_key] = item.pop(key)
+
+        # Write to file
+        data = response_json["data"]
+        record_count = len(data)
+        df = pd.DataFrame(data)
+        df.to_parquet(output_parquet_filename.format(page_number))
+        total_count += record_count
+
+        # Check if there are more records to retrieve
+        if record_count < 2000:
+            break  # Stop if there are no more records
+
+        # Increment the offset for the next page
+        request_payload["query"]["offset"] += 2000
+        page_number += 1
+
+        # Add a 5-second delay before fetching the next page
+        time.sleep(5)
+
+    logger.info(f"Total records retrieved: {total_count}")
+
+
+table_schema = """
+    record_date DATE,
+    network STRING,
+    url STRING,
+    utm_campaign STRING,
+    utm_content STRING,
+    utm_medium STRING,
+    utm_source STRING,
+    utm_term STRING,
+    site_domain STRING,
+    ad_unit STRING,
+    size STRING,
+    country_code STRING,
+    device_type STRING,
+    device_os STRING,
+    browser STRING,
+    integration_partner STRING,
+    impressions INTEGER,
+    net_revenue INTEGER,
+    net_cpm FLOAT
+"""
+
+# Define table to load
+snowflake_table = "freestar_daily_extracts"
+
+# Define SQL statements
+create_schema_sql = """
+CREATE SCHEMA IF NOT EXISTS freestar;
+"""
+
+create_table_sql = f"""
+CREATE TABLE IF NOT EXISTS freestar.{snowflake_table} (
+    {table_schema}
+);
+"""
+
+format_file_sql = f"""
+CREATE OR REPLACE FILE FORMAT {snowflake_table}_format
+TYPE = parquet;
+"""
+
+create_stage_sql = f"""
+CREATE OR REPLACE STAGE {snowflake_table}_stage
+FILE_FORMAT = {snowflake_table}_format;
+"""
+
```

Comment:
Any specific advantage of adding the DDLs with the DMLs?

Reply:
will discuss next week